bundles/org.openhab.binding.elroconnects/src/main/java/org/openhab/binding/elroconnects/internal/handler/ElroConnectsDeviceHandler.java

```diff
@@ -56,10 +59,22 @@ public void initialize() {
         }
 
         if (bridgeHandler.getThing().getStatus() == ThingStatus.ONLINE) {
-            bridgeHandler.setDeviceHandler(deviceId, this);
-            updateProperties(bridgeHandler);
-            updateDeviceName(bridgeHandler);
-            refreshChannels(bridgeHandler);
+            ElroConnectsDevice device = bridgeHandler.getDevice(deviceId);
+            if (device != null) {
+                ElroDeviceType deviceType = TYPE_MAP.get(device.getDeviceType());
+                if ((deviceType == null) || !thing.getThingTypeUID().equals(THING_TYPE_MAP.get(deviceType))) {
+                    String msg = String.format("@text/offline.invalid-device-type [ \"%d\" ]", deviceType);
```

Comment:
Take care, deviceType  is apparently not a number and it can be also null.

Reply:
String.format will not throw an exception if deviceType is null ?
I do not remember if String.format handles properly null parameters.

Reply:
Indeed, should be %s in this case.
deviceType can indeed be null if it is not found in the TYPE_MAP. That would be an issue with the binding not knowing the device type. At the moment, the binding maps all device types also known by the official app. I considered separately logging it but decided to keep it simple. The formatter will just show null in the string. This can anyhow only happen when a user tries to manually configure, something which is not obvious, because the deviceId can only be found through discovery, or console commands. And it is a required parameter.

Reply:
Ok

Reply:
I checked the javadoc, and String.format should write out null if the value is null.
Anyway, checking again, the value can actually not be null here. The bridge handler will not put the device in its devices map when the type is not known in the binding. So it should never be null. The only reason to keep the null check is for null type checking (get can return null in theory).
So my last fix is not required, and I will actually revert it. The type enum is more meaningful and easier to understand than the 4 digit hex codes behind it.